### PR TITLE
refactor(GUI): make OSWindowProgress a simple CommonJS module

### PR DESCRIPTION
**Pull request by jviotti**
There is no need to tie this module to Angular by using an Angular
service.

Signed-off-by: Juan Cruz Viotti <jviotti@openmailbox.org>

### CODE REVIEW

File: lib/gui/os/window-progress.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * @param {Number} percentage - percentage
+ *
+ * @example
+ * windowProgress.set(85);
+ */
+exports.set = (percentage) => {
+  const PERCENTAGE_MINIMUM = 0;
+  const PERCENTAGE_MAXIMUM = 100;
+
+  if (percentage > PERCENTAGE_MAXIMUM || percentage < PERCENTAGE_MINIMUM) {
```
Comment (lurch) [inline lib/gui/os/window-progress.js]: There now seems to be several places throughout Etcher where we're checking that a percentage lies within the 0 -> 100 range, which leads to many re-definitions of `ZERO` and `HUNDRED`. Maybe it's time to add a shared `isValidPercentage` function?

File: lib/gui/os/window-progress.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    throw errors.createError(`Invalid window progress percentage: ${percentage}`);
+  }
+
+  exports.currentWindow.setProgressBar(percentage / PERCENTAGE_MAXIMUM);
+};
+
+/**
+ * @summary Clear the operating system window progress bar
```
Comment (lurch) [inline lib/gui/os/window-progress.js]: "operating system window progress bar" feels like a bit of a mouthful (and a bit vague). Is there a more succinct name for this?

**Comment by lurch** [comment]
Requested a review from @Shou because I guess it ties up with some of the Angular-removal work he's also been doing :-)

File: lib/gui/os/window-progress.js (hunk 1 above)
Comment (jviotti) [inline lib/gui/os/window-progress.js]: Done!

File: lib/gui/os/window-progress.js (hunk 2 above)
Comment (jviotti) [inline lib/gui/os/window-progress.js]: I reduced it to "Clear the window progress bar."

File: tests/shared/utils.spec.js [hunk 3]
```diff
@@ -333,4 +333,32 @@ describe('Shared: Utils', function() {
 
   });
 
+  describe('.isValidPercentage()', function() {
```
Comment (lurch) [inline tests/shared/utils.spec.js]: Could also test it with floating-point numbers? E.g. -0.1, 0.1, 12.3, 99.9, 100.1

Comment (jviotti) [inline tests/shared/utils.spec.js]: Done!

File: lib/gui/os/window-progress.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+ *
+ * @example
+ * windowProgress.set(85);
+ */
+exports.set = (percentage) => {
+  if (!utils.isValidPercentage(percentage)) {
+    throw errors.createError(`Invalid window progress percentage: ${percentage}`);
+  }
+
+  const PERCENTAGE_MAXIMUM = 100;
```
Comment (lurch) [inline lib/gui/os/window-progress.js]: I wonder if this constant could / should be exposed from `utils`?

Or perhaps there could be a `utils.percentageToFloat` function? \*shrug\*

File: lib/shared/utils.js [hunk 5]
```diff
@@ -133,3 +133,26 @@ exports.hideAbsolutePathsInObject = (object) => {
...
+ *
+ * @param {Number} percentage - percentage
+ * @returns {Boolean} whether the percentage is valid
+ *
+ * @example
+ * if (utils.isValidPercentage(85)) {
+ *   console.log('The percentage is valid');
+ * }
+ */
+exports.isValidPercentage = (percentage) => {
```
Comment (lurch) [inline lib/shared/utils.js]: :tada:

File: lib/gui/os/window-progress.js (hunk 4 above)
Comment (jviotti) [inline lib/gui/os/window-progress.js]: That's a great idea. Implemented!

File: tests/shared/utils.spec.js [hunk 6]
```diff
@@ -333,4 +333,88 @@ describe('Shared: Utils', function() {
...
+      m.chai.expect(utils.isValidPercentage(101)).to.be.false;
+    });
+
+    it('should return false if percentage is a float greater than 100', function() {
+      m.chai.expect(utils.isValidPercentage(100.001)).to.be.false;
+    });
+
+  });
+
+  describe('.percentageToFloat()', function() {
```
Comment (lurch) [inline tests/shared/utils.spec.js]: No test for passing in strings? ;)

Comment (jviotti) [inline tests/shared/utils.spec.js]: Done!

**Comment by jviotti** [comment]
Oh, this PR has now tons of conflicts :(

**Comment by lurch** [comment]
Re-pinging @Shou for a review.

File: tests/shared/utils.spec.js (hunk 6 above)
Comment (Shou) [inline tests/shared/utils.spec.js]: Maybe `null` would be good to test too considering `null <= 100` is `true`, to make it explicit we avoid this pitfall?

File: lib/shared/utils.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+ * @returns {Number} float percentage
+ *
+ * @example
+ * const value = utils.percentageToFloat(50);
+ * console.log(value);
+ * > 0.5
+ */
+exports.percentageToFloat = (percentage) => {
+  if (!exports.isValidPercentage(percentage)) {
+    throw errors.createError(`Invalid window progress percentage: ${percentage}`);
```
Comment (Shou) [inline lib/shared/utils.js]: Also, we should make this error generic instead of referencing the window progress.

Comment (Shou) [inline lib/shared/utils.js]: Also, we should make this error generic instead of referencing the window progress.

Comment (lurch) [inline lib/shared/utils.js]: Good spot!

File: tests/shared/utils.spec.js (hunk 6 above)
Comment (jviotti) [inline tests/shared/utils.spec.js]: Oh, that's some hardcore JavaScript weirdness :P Fixed!

File: lib/shared/utils.js (hunk 7 above)
Comment (jviotti) [inline lib/shared/utils.js]: Oh, right, good catch!

File: tests/shared/utils.spec.js [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+    it('should throw an error if given a string percentage', function() {
+      m.chai.expect(function() {
+        utils.percentageToFloat('50');
+      }).to.throw('Invalid percentage: 50');
+    });
+
+    it('should throw an error if given a null percentage', function() {
```
Comment (lurch) [inline tests/shared/utils.spec.js]: IMHO you should add this test to the `.isValidPercentage()` section too.

Comment (jviotti) [inline tests/shared/utils.spec.js]: Done!

**Pull request merged**